src_c/math.c

```diff
@@ -1287,6 +1293,99 @@ vector_scale_to_length(pgVector *self, PyObject *length)
     Py_RETURN_NONE;
 }
 
+static int
+_vector_move_towards_helper(Py_ssize_t dim, double *origin_coords, double *target_coords, double distance)
+{
+    Py_ssize_t i;
+    double delta[VECTOR_MAX_SIZE];
+    double dist;
+
+    if (distance < 0) {
+        PyErr_SetString(PyExc_TypeError, 
+            "Expected distance to be greater or equal to 0");
+        return 0;
+    }
+
+    for (i = 0; i < dim; ++i)
+        delta[i] = origin_coords[i] - target_coords[i];
+
+    /* Get magnitude of Vector */
+    dist = sqrt(_scalar_product(delta, delta, dim));
+
+    if (dist <= distance || dist == 0)
```

Comment:
would make more logical sense if `target` Vector is returned when `dist <= distance`, but when `dist == 0`, the `origin` vector should be left as is, and returned